homeassistant/components/downloader/__init__.py

```diff
@@ -195,7 +194,7 @@ def do_download() -> None:
                 if final_path and os.path.isfile(final_path):
                     os.remove(final_path)
 
-        threading.Thread(target=do_download).start()
+        await hass.async_add_executor_job(do_download)
 
     async_register_admin_service(
```

Comment:
Side note: Integration services should be registered in `async_setup` and not during config entry setup.